system/jlib/jptree.cpp

```diff
@@ -2869,36 +2910,44 @@ AttrValue *PTree::getNextAttribute(AttrValue *cur) const
 
 // LocalPTree
 
-LocalPTree::LocalPTree(const char *_name, byte _flags, IPTArrayValue *_value, ChildMap *_children) : PTree(_flags, _value, _children)
+std::atomic<unsigned> localTrees;
+
+LocalPTree::LocalPTree(const char *_name, byte _flags, IPTArrayValue *_value, ChildMap *_children) : PTree(_flags|ipt_fast, _value, _children)
 {
     if (_name)
         setName(_name);
+    localTrees++;
```

Comment:
efficiency: Should use localTrees.fetch_add(1, std::memory_order_relaxed);

since only requirement is atomicity, rather than consistent ordering.  (E.g., see example on http://en.cppreference.com/w/cpp/atomic/memory_order)

Same goes fro decrement.
You could use relaxedAtomic<x> instead (used by code in jfile.cpp for ioReadCycles).